pulser-core/pulser/json/abstract_repr/deserializer.py

```diff
@@ -40,51 +41,70 @@
     Waveform,
 )
 
-if TYPE_CHECKING:
+if TYPE_CHECKING:  # pragma: no cover
     from pulser.sequence import Sequence
 
-with open("pulser-core/pulser/json/abstract_repr/schema.json") as f:
+with open(Path(__file__).parent / "schema.json") as f:
     schema = json.load(f)
 
-
 VARIABLE_TYPE_MAP = {"int": int, "float": float}
 
+ExpReturnType = Union[int, float, ParamObj]
+
+
+@overload
+def _deserialize_parameter(param: int, vars: dict[str, Variable]) -> int:
```

Comment:
All those lines just for mypy...

Reply:
Hahaha indeed, it's very high maintenance